kyaml/fn/framework/command/example_test.go

```diff
@@ -0,0 +1,383 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package command_test
+
+import (
+	"bytes"
+	"fmt"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const service = "Service"
+
+// ExampleBuild_modify implements a function that sets an annotation on each resource.
+// The annotation value is configured via ResourceList.FunctionConfig.
+func ExampleBuild_modify() {
+	// create a struct matching the structure of ResourceList.FunctionConfig to hold its data
+	config := &struct {
+		Data map[string]string `yaml:"data" json:"data"`
+	}{}
+	fn := func(items []*yaml.RNode) ([]*yaml.RNode, error) {
+		for i := range items {
+			// set the annotation on each resource item
+			err := items[i].PipeE(yaml.SetAnnotation("value", config.Data["value"]))
+			if err != nil {
+				return nil, err
+			}
+		}
+		return items, nil
+	}
+
+	cmd := command.Build(framework.SimpleProcessor{Filter: kio.FilterFunc(fn), Config: config},
+		true, false)
+
+	// for testing purposes only -- normally read from stdin when Executing
+	cmd.SetIn(bytes.NewBufferString(`
+apiVersion: config.kubernetes.io/v1alpha1
+kind: ResourceList
+# items are provided as nodes
+items:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: foo
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: foo
+functionConfig:
+  apiVersion: v1
+  kind: ConfigMap
+  data:
+    value: baz
+`))
+	// run the command
+	if err := cmd.Execute(); err != nil {
+		panic(err)
+	}
+
+	// Output:
+	// apiVersion: config.kubernetes.io/v1alpha1
+	// kind: ResourceList
+	// items:
+	// - apiVersion: apps/v1
+	//   kind: Deployment
+	//   metadata:
+	//     name: foo
+	//     annotations:
+	//       value: 'baz'
```

Comment:
I think there's a parsing error somewhere if the value ends up being `'baz'` instead of `baz`

Reply:
As I suspected above, it is the AnnotationSetter that is doing this. It explains why:

```go
	// some tools get confused about the type if annotations are not quoted
	v.YNode().Style = yaml.SingleQuotedStyle
```

Reply:
See comment above regarding annotations.